SSA/Projects/InstCombine/Base.lean

```diff
@@ -88,6 +88,22 @@ deriving Repr, DecidableEq, Inhabited
 
 /-- Homogeneous, binary operations -/
 inductive MOp.BinaryOp : Type
+  | and
+  | or
+  | xor
+  | shl
+  | lshr
+  | ashr
+  | urem
+  | srem
+  | add (nswnuw  : Bool × Bool := ⟨ false , false ⟩ )
+  | mul
+  | sub
+  | sdiv
+  | udiv
+deriving DecidableEq, Inhabited
+/-- Homogeneous, binary operations -/
+inductive g : Type
```

Comment:
`g` is not a very descriptive name, which is not helped by the fact that you merely copied the docstring!
It seems to be a copy of `MOp` without the overflow flags. Document this connection, and explain why we need it.